### PR TITLE
Add The Boulevard keyboard

### PR DESCRIPTION
**Pull request by joedinkle**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [X ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by joedinkle** [comment]
Corrected checklist:

[X ] My code follows the code style of this project: C, Python
[ X] I have read the PR Checklist document and have made the appropriate changes.
[ ] My change requires a change to the documentation.
[ ] I have updated the documentation accordingly.
[ X] I have read the CONTRIBUTING document.
[ ] I have added tests to cover my changes.
[ X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

File: keyboards/eyeohdesigns/theboulevard/config.h [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+#define DIODE_DIRECTION COL2ROW
+
+#define RGB_DI_PIN F6
+#define RGBLED_NUM 2
+#define RGBLIGHT_HUE_STEP 8
+#define RGBLIGHT_SAT_STEP 8
+#define RGBLIGHT_VAL_STEP 8
+#define RGBLIGHT_LIMIT_VAL 255 /* The maximum brightness level */
+#define RGBLIGHT_SLEEP  /* If defined, the RGB lighting will be switched off when the host goes to sleep */
+#define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/eyeohdesigns/theboulevard/config.h:51]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/eyeohdesigns/theboulevard/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+bool encoder_update_user(uint8_t index, bool clockwise) {
+  if (index == 0) {
+    if (clockwise) {
+      tap_code(KC_VOLU);
+    }  else{
+      tap_code(KC_VOLD);
+    }
+  }
+  return true;
+}
```
Comment (drashna) [inline keyboards/eyeohdesigns/theboulevard/keymaps/default/keymap.c:59]:
```suggestion
```

File: keyboards/eyeohdesigns/theboulevard/keymaps/ortho2/keymap.c [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+bool encoder_update_user(uint8_t index, bool clockwise) {
+  if (index == 0) {
+    if (clockwise) {
+      tap_code(KC_VOLU);
+    }  else{
+      tap_code(KC_VOLD);
+    }
+  }
+  return true;
+}
```
Comment (drashna) [inline keyboards/eyeohdesigns/theboulevard/keymaps/ortho2/keymap.c:59]:
```suggestion
```

File: keyboards/eyeohdesigns/theboulevard/keymaps/default/keymap.c (hunk 2 above)
Comment (drashna) [inline keyboards/eyeohdesigns/theboulevard/keymaps/default/keymap.c:59]:
```suggestion
```

File: keyboards/eyeohdesigns/theboulevard/keymaps/ortho4/keymap.c [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+bool encoder_update_user(uint8_t index, bool clockwise) {
+  if (index == 0) {
+    if (clockwise) {
+      tap_code(KC_VOLU);
+    }  else{
+      tap_code(KC_VOLD);
+    }
+  }
+  return true;
+}
```
Comment (drashna) [inline keyboards/eyeohdesigns/theboulevard/keymaps/ortho4/keymap.c:59]:
```suggestion
```

File: keyboards/eyeohdesigns/theboulevard/keymaps/ortho3/keymap.c [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+bool encoder_update_user(uint8_t index, bool clockwise) {
+  if (index == 0) {
+    if (clockwise) {
+      tap_code(KC_VOLU);
+    }  else{
+      tap_code(KC_VOLD);
+    }
+  }
+  return true;
+}
```
Comment (drashna) [inline keyboards/eyeohdesigns/theboulevard/keymaps/ortho3/keymap.c:59]:
```suggestion
```

File: keyboards/eyeohdesigns/theboulevard/keymaps/ortho5/keymap.c [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+bool encoder_update_user(uint8_t index, bool clockwise) {
+  if (index == 0) {
+    if (clockwise) {
+      tap_code(KC_VOLU);
+    }  else{
+      tap_code(KC_VOLD);
+    }
+  }
+  return true;
+}
```
Comment (drashna) [inline keyboards/eyeohdesigns/theboulevard/keymaps/ortho5/keymap.c:59]:
```suggestion
```

File: keyboards/eyeohdesigns/theboulevard/keymaps/stagger1/keymap.c [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+bool encoder_update_user(uint8_t index, bool clockwise) {
+  if (index == 0) {
+    if (clockwise) {
+      tap_code(KC_VOLU);
+    }  else{
+      tap_code(KC_VOLD);
+    }
+  }
+  return true;
+}
```
Comment (drashna) [inline keyboards/eyeohdesigns/theboulevard/keymaps/stagger1/keymap.c:59]:
```suggestion
```

File: keyboards/eyeohdesigns/theboulevard/keymaps/stagger2/keymap.c [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+bool encoder_update_user(uint8_t index, bool clockwise) {
+  if (index == 0) {
+    if (clockwise) {
+      tap_code(KC_VOLU);
+    }  else{
+      tap_code(KC_VOLD);
+    }
+  }
+  return true;
+}
```
Comment (drashna) [inline keyboards/eyeohdesigns/theboulevard/keymaps/stagger2/keymap.c:59]:
```suggestion
```

File: keyboards/eyeohdesigns/theboulevard/keymaps/ortho3/keymap.c (hunk 5 above)
Comment (drashna) [inline keyboards/eyeohdesigns/theboulevard/keymaps/ortho3/keymap.c:59]:
```suggestion
```

File: keyboards/eyeohdesigns/theboulevard/keymaps/stagger4/keymap.c [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+bool encoder_update_user(uint8_t index, bool clockwise) {
+  if (index == 0) {
+    if (clockwise) {
+      tap_code(KC_VOLU);
+    }  else{
+      tap_code(KC_VOLD);
+    }
+  }
+  return true;
+}
```
Comment (drashna) [inline keyboards/eyeohdesigns/theboulevard/keymaps/stagger4/keymap.c:59]:
```suggestion
```

File: keyboards/eyeohdesigns/theboulevard/keymaps/stagger5/keymap.c [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+bool encoder_update_user(uint8_t index, bool clockwise) {
+  if (index == 0) {
+    if (clockwise) {
+      tap_code(KC_VOLU);
+    }  else{
+      tap_code(KC_VOLD);
+    }
+  }
+  return true;
+}
```
Comment (drashna) [inline keyboards/eyeohdesigns/theboulevard/keymaps/stagger5/keymap.c:59]:
```suggestion
```

File: keyboards/eyeohdesigns/theboulevard/rules.mk [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
```
Comment (drashna) [inline keyboards/eyeohdesigns/theboulevard/rules.mk:17]:
```suggestion
```

File: keyboards/eyeohdesigns/theboulevard/rules.mk [hunk 12]
```diff
@@ -0,0 +1,20 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
```
Comment (drashna) [inline keyboards/eyeohdesigns/theboulevard/rules.mk:10]:
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
```

File: keyboards/eyeohdesigns/theboulevard/theboulevard.c [hunk 13]
```diff
@@ -0,0 +1,17 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "theboulevard.h"
```
Comment (drashna) [inline keyboards/eyeohdesigns/theboulevard/theboulevard.c:17]: This allows for a default behavior to be defined, without the need for the user to specify the behavior.  This is very useful for the configurator which doesn't have a way to configure this behavior, currently.

```suggestion
#include "theboulevard.h" 

#ifdef ENCODER_ENABLE
bool encoder_update_kb(uint8_t index, bool clockwise) {
    if (!encoder_update_user(index, clockwise)) { return false; }
    if (index == 0) {
        if (clockwise) {
            tap_code_delay(KC_VOLU, 10);
        } else {
            tap_code_delay(KC_VOLD, 10);
        }
    }
    return true;
}
#endif
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/eyeohdesigns/theboulevard/keymaps/default/keymap.c (hunk 2 above)
Comment (joedinkle) [reply keyboards/eyeohdesigns/theboulevard/keymaps/default/keymap.c]: I committed all of the changes I could. The option to commit for the default keymap is not enabled, but it looks like the changes took.

File: keyboards/eyeohdesigns/theboulevard/config.h [hunk 14]
```diff
@@ -0,0 +1,53 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID 0xFEED
+#define PRODUCT_ID 0x0000
+#define DEVICE_VER 0x0001
+#define MANUFACTURER eye oh designs
+#define PRODUCT theboulevard
+#define DESCRIPTION battle station style keeb
```
Comment (drashna) [inline keyboards/eyeohdesigns/theboulevard/config.h:28]:
```suggestion
```

File: keyboards/eyeohdesigns/theboulevard/keymaps/stagger3/keymap.c [hunk 15]
```diff
@@ -0,0 +1,59 @@
...
+bool encoder_update_user(uint8_t index, bool clockwise) {
+  if (index == 0) {
+    if (clockwise) {
+      tap_code(KC_VOLU);
+    }  else{
+      tap_code(KC_VOLD);
+    }
+  }
+  return true;
+}
```
Comment (drashna) [inline keyboards/eyeohdesigns/theboulevard/keymaps/stagger3/keymap.c:59]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/eyeohdesigns/theboulevard/rules.mk [hunk 16]
```diff
@@ -0,0 +1,17 @@
...
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+NKRO_ENABLE = no            # USB Nkey Rollover
+RGBLIGHT_ENABLE = yes        # Enable keyboard RGB underglow
```
Comment (fauxpark) [inline keyboards/eyeohdesigns/theboulevard/rules.mk:16]:
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
NKRO_ENABLE = no            # Enable N-Key Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by joedinkle** [comment]
Requested changes made.

**Review by fauxpark — APPROVED** [review]

**Comment by joedinkle** [comment]
Now that things are approved, would you fine folks mind giving this a merge? Thanks!

**Pull request merged**